README.md

```diff
@@ -130,19 +156,242 @@ Usage: txid2txref [options] <txid>
  --rpcpassword [pass]       RPC password
  --rpcport [port]           RPC port (default: try both 8332 and 18332)
  --config [config_path]     Full pathname to bitcoin.conf (default: <homedir>/.bitcoin/bitcoin.conf)
+ --utxoIndex [index #]      Index # for UXTO within the transaction (default: 0)
+ --extended                 Force output of a txref-ext
+
+<txid|txref|txref-ext>      input: can be a txid to encode, or a txref or txref-ext to decode
+```
 
-$ # specify all credentials
-$ ./src/txid2txref --rpcuser bitcoinrpc --rpcpassword super-secret-passwd --rpchost 127.0.0.1 --rpcport 18333 f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107
-txid: f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107
-txref: txtest1-xyv2-xzyq-qqm5-tyke
+Many of the runtime options for txid2txref are for connecting to
+bitcoind over RPC. There are four `--rpc*` options that you can use for
+connecting to a local or remote bitcoind:
+
+```
+$ ./src/txid2txref --rpcuser bitcoinrpc --rpcpassword super-secret-passwd --rpchost 127.0.0.1 --rpcport 18332 <txid>
+{...}
+```
+
+If you have bitcoind running locally, you probably have a bitcoin.conf
+file with all the RPC info in there. If so, you can use it instead of
+setting all of the `--rpc*` options:
+
+```
+$ # by default it will look in <homedir>/.bitcoin/bitcoin.conf
+$ ./src/txid2txref <txid>
+{...}
+
+$ # or you can use a different location
+$ ./src/txid2txref --config /tmp/bitcoin.conf <txid>
+{...}
+```
+
+When given a txid, txid2txref will connect to bitcoind to verify it is a
+valid txid and, if found, gather some info about it. Then it will output
+some JSON with details about that transaction, and the txref.
+
+```
+$ ./src/txid2txref f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7521234
+Error: transaction f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7521234 not found.
 
-$ # no credentials...will look in <homedir>/.bitcoin/bitcoin.conf
 $ ./src/txid2txref f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107
-txid: f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107
-txref: txtest1-xyv2-xzyq-qqm5-tyke
+{
+  "txid": "f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107",
+  "txref": "txtest1-xyv2-xzyq-qqpq-q6k0a23",
+  "network": "test",
+  "block-height": "1152194",
+  "transaction-position": "1",
+  "utxo-index": "1",
+  "query-string": "f8cdaff3ebd9e862ed5885f8975489090595abe1470397f79780ead1c7528107"
+}
+```
+
+txid2txref can also output an extended txref if given a utxoIndex > 0, or by using the flag
```

Comment:
Is --extended using txo #0?

Reply:
Yes, I struggled with how best to make this work. txref "standard" does not reference a txo, so txid2txref considers the default txo to be #0. If the user gives a non-zero # with --utxoIndex, then we of course output a txref-ext. However, it might be that the user wants a txref-ext even if the txo is really #0, so I had to add an extra flag to get that to happen, --extended. 

Depending where we want to go, we could remove the --extended option, make txref-ext output the default operation, and only output "standard" txrefs IF the txo is #0 and some new option, like --short, is specified? ... somewhat annoying either way.

Reply:
I just thought of a third option: output standard-size txref UNLESS --utxoIndex is given, zero or non-zero, then output txref-ext.

Also, I should change --utxoIndex to --txoIndex.

Reply:
Yes, I think short as default solves this best.